src/benchmarks/gc/GC.Infrastructure/GC.Analysis.API/Common.cs

```diff
@@ -4,6 +4,70 @@
 using System.Text;
 using XPlot.Plotly;
 
+namespace System.Linq
+{
+    public static class Linq
+    {
+        public static List<R> EagerSelect<T, R>(this IEnumerable<T> data, Func<T, R> map)
+        {
+            List<R> result = new();
+            foreach (var d in data)
+            {
+                result.Add(map(d));
+            }
+
+            return result;
+        }
+
+        public static List<T> EagerWhere<T>(this IEnumerable<T> data, Func<T, bool> predicate)
+        {
+            List<T> result = new();
+            foreach (var d in data)
+            {
+                if (predicate(d))
+                {
+                    result.Add(d);
+                }
+            }
+
+            return result;
+        }
+
+        public static double EagerSum<TSource>(this IEnumerable<TSource> data, Func<TSource, double> map)
+        {
+            double sum = 0;
+            if (data == null || data.Count() == 0) 
+            {
+                return sum;
+            }
+
+            foreach(var sourceItem in data)
+            {
+                sum += (double)map(sourceItem);
+            }
+
+            return sum;
+        }
+
+        public static double EagerAverage<TSource>(this IEnumerable<TSource> data, Func<TSource, double> map)
+        {
+            int count = data.Count();
+            if (data == null || count == 0) 
+            {
+                return double.NaN;
+            }
+
+            double sum = 0;
+            foreach(var sourceItem in data)
+            {
+                sum += (double)map(sourceItem);
+            }
+
+            return sum / count;
+        }
+    }
+}
```

Comment:
The standard Linq methods are already eager for these two.  These differ from Linq
- in their handling of null (no exception)
- Average(<empty>) returning NaN instead of throwing.  (though there are Linq overloads that takes enumerables of nullables, and if the enumerable has no nonnull values (empty enumerable is a degenerate case of this) returns `null`)
- calling `Count()` rather than counting during the `foreach` - if the enumerable is a List this won't matter, but for others it could enumerate twice
- Linq uses checked arithmetic
-